[quant] Refactor Quantizer to remove the Affine keyword

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #70324
* __->__ #70323

Summary:
the current Quantizer only uses the Affine keyword even though it may be used for quantized tensors created using affine/symmetric qscheme.
This PR refactors the code to remove the affine keyword from the class name and make it accept qscheme as input.
Future PRs will update the callsites for creating affine/symmetric quantized tensor.

Addresses https://github.com/pytorch/pytorch/issues/51152
Test Plan:
python test/test_quantization.py

Reviewers:

Subscribers:

Tasks:

Tags:

supriyar committed:
[quant] Refactor Quantizer to remove the Affine keyword

Summary:
the current Quantizer only uses the Affine keyword even though it may be used for quantized tensors created using affine/symmetric qscheme.
This PR refactors the code to remove the affine keyword from the class name and make it accept qscheme as input.
Future PRs will update the callsites for creating affine/symmetric quantized tensor.

Test Plan:
python test/test_quantization.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on aten/src/ATen/quantized/Quantizer.h:
```diff
@@ -108,19 +95,21 @@ struct TORCH_API PerTensorAffineQuantizer : public AffineQuantizer {
         zero_points_(zero_points),
-        axis_(axis) {}
+        axis_(axis),
+        qscheme_(qscheme) {}
```
do we want to do some validation for qscheme?

jerryzh168 commented on aten/src/ATen/quantized/Quantizer.h:
```diff
@@ -171,13 +161,13 @@ struct TORCH_API PerChannelAffineQuantizer : public AffineQuantizer {
       Tensor zero_points,
       int64_t axis)
-      : PerChannelAffineQuantizer(scalar_type,
+      : PerChannelQuantizer(scalar_type,
```
is there a symmetric/affine variant for this quantizer?

supriyar replied:
No, this one only has affine variant.

jerryzh168 reviewed (approved):
looks good, had some comments inline

supriyar committed:
Update on "[quant] Refactor Quantizer to remove the Affine keyword"


Summary:
the current Quantizer only uses the Affine keyword even though it may be used for quantized tensors created using affine/symmetric qscheme.
This PR refactors the code to remove the affine keyword from the class name and make it accept qscheme as input.
Future PRs will update the callsites for creating affine/symmetric quantized tensor.

Addresses https://github.com/pytorch/pytorch/issues/51152
Test Plan:
python test/test_quantization.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]
